data-node/src/main/java/org/graylog/datanode/bootstrap/preflight/OpensearchDataDirCompatibilityCheck.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog.datanode.bootstrap.preflight;
+
+import org.graylog.datanode.configuration.DatanodeConfiguration;
+import org.graylog.datanode.filesystem.index.IncompatibleIndexVersionException;
+import org.graylog.datanode.filesystem.index.IndicesDirectoryParser;
+import org.graylog.datanode.filesystem.index.dto.IndexerDirectoryInformation;
+import org.graylog.datanode.filesystem.index.dto.NodeInformation;
+import org.graylog.shaded.opensearch2.org.opensearch.Version;
+import org.graylog2.bootstrap.preflight.PreflightCheck;
+import org.graylog2.bootstrap.preflight.PreflightCheckException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import java.nio.file.Path;
+import java.util.Locale;
+
+public class OpensearchDataDirCompatibilityCheck implements PreflightCheck {
+
+    private final boolean isFreshInstallation;
+    private final DatanodeConfiguration datanodeConfiguration;
+    private final IndicesDirectoryParser indicesDirectoryParser;
+
+    @Inject
+    public OpensearchDataDirCompatibilityCheck(@Named("isFreshInstallation") boolean isFreshInstallation, DatanodeConfiguration datanodeConfiguration, IndicesDirectoryParser indicesDirectoryParser) {
```

Comment:
I think we may have a problem with `isFreshInstallation` .
If I start from scratch and boot up a Graylog Server first and then a Data Node, `isFreshInstallation` will be set to false already since the PreflightCheck only checks for the `org.graylog2.plugin.cluster.ClusterId` in `cluster_config`.
I think we need something equivalent for data node only.

Reply:
Never mind. It does work as expected. Sorry for the confusion